### ISSUE TITLE
Files without contracts cause the Truffle plugin to fail

### ISSUE THREAD

**Issue by frangio** [OpenZeppelin/openzeppelin-upgrades#826]
As reported [in the forum](https://forum.openzeppelin.com/t/no-node-with-id-645-of-type-contractdefinition/36705), using the Truffle plugin with the following contract fails:

```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.17;

import "operator-filter-registry/src/DefaultOperatorFilterer.sol";

contract Vox is DefaultOperatorFilterer {}
```

The error is something like `Error: No node with id 645 of type ContractDefinition`.

---

I traced this issue to this line where the AST for the `DefaultOperatorFilterer` contract is deleted from the reconstructed compiler output.:

https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/fc55c0dc8a992f649ec415f47797d5b5c8088b0c/packages/plugin-truffle/src/utils/validations.ts#L150

This is roughly due to:
https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/fc55c0dc8a992f649ec415f47797d5b5c8088b0c/packages/plugin-truffle/src/utils/validations.ts#L117

Specifically, because of the [Constants.sol](https://github.com/ProjectOpenSea/operator-filter-registry/blob/12dde4c80d49af122db5380b203bc202011e8c9b/src/lib/Constants.sol) file which doesn't have a contract in it.

tl;dr: A heuristic we use is failing.

I realize now that we could use a much more direct way to recognize contracts that need to be deleted from the reconstructed compiler output. We should find all contracts that are reachable through imports from files in the user's own contracts directory, and delete anything else.
